### PR TITLE
Add sphinx-github-changelog

### PR DESCRIPTION
**Pull request by ewjoachim**
Add https://github.com/ewjoachim/sphinx-github-changelog / https://pypi.org/project/sphinx-github-changelog

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by curtisma** [comment]
@ewjoachim Yeah, sure!

**Comment by curtisma** [comment]
@ewjoachim, would you expect this to build on all platforms?

I can take a look at the build failures and make some updates but not until at least this evening.

**Comment by ewjoachim** [comment]
I... Don't really know :D

It fails with `ModuleNotFoundError: No module named 'poetry'`, and I reproduce locally with `CONFIG=linux64 ./.scripts/run_docker_build.sh`, but I haven't looked close enough at why.

**Comment by ewjoachim** [comment]
A few notes:
- I guess it's missing somewhere a place where poetry is listed (`requirements > host` maybe ?)
- I don't know if I need to redefine my dependencies here or if they are extracted from the package metadata.

**Comment by curtisma** [comment]
I think you need to add "poetry" to the test-> requirements section of the meta.yaml

`test:
  # Some packages might need a `test/commands` key to check CLI.
  # List all the packages/modules that `run_test.py` imports.
  imports:
    - sphinx_github_changelog
  # For python packages, it is useful to run pip check. However, sometimes the
  # metadata used by pip is out of date. Thus this section is optional if it is
  # failing.
  source_files:
    - tests
  requires:
    - pip
    - poetry
  commands:
    - pip check
    - poetry install
    - poetry run pytest`

**Comment by ewjoachim** [comment]
Ah I must have pushed the wrong commit, I had that locally and it failed identically. Let me update the commit :)

**Comment by ewjoachim** [comment]
New error:
> sphinx 4.4.0 has requirement importlib-metadata>=4.4; python_version < "3.10", but you have importlib-metadata 3.10.1.

I'm going to fix the lib underneath

**Comment by ewjoachim** [comment]
There's one thing I'm wondering: if I want the tests to run, do I need to duplicate the code of the tests between this repo and my repo ? If so I think I'd rather avoid this (except if we can automate it). If there's a way to tell Conda where my tests are located, then let's do that instead. Note: I'd rather not distribute the tests with the package (except if people really think I should)

**Comment by ewjoachim** [comment]
Looks like we're starting to see some green :)

There's one thing I'm wondering: if I want the tests to run, do I need to duplicate the code of the tests between this repo and my repo ? If so I think I'd rather avoid this (except if we can automate it). If there's a way to tell Conda where my tests are located, then let's do that instead. Note: I'd rather not distribute the tests with the package (except if people really think I should)

@conda-forge/help-python ?

**Comment by ewjoachim** [comment]
@conda-forge-admin, please ping conda-forge/help-python

**Comment by ewjoachim** [comment]
I don't know if there's something specific I should be doing

**Comment by curtisma** [comment]
> I don't know if there's something specific I should be doing

hmm, I think we might need to tag the admins asking them to merge this PR.  Then that will make a new "feedstock" repo containing the recipe we submitted and add us to it.

**Comment by curtisma** [comment]
@conda-forge-admin, please merge this PR

File: recipes/sphinx-github-changelog/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    # - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - poetry-core
+  run:
+    - docutils
+    - importlib-metadata  # [py<38]
```
Comment (xhochy) [inline recipes/sphinx-github-changelog/meta.yaml:48]: We can add this unconditionally to make it noarch

```suggestion
    - importlib-metadata
```

File: recipes/sphinx-github-changelog/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  # url: https://github.com/simplejson/simplejson/releases/download/{{ version }}/simplejson-{{ version }}.tar.gz
+  # and otherwise fall back to archive:
+  # url: https://github.com/simplejson/simplejson/archive/v{{ version }}.tar.gz
+  sha256: 1104d66fcb1849e46637e9b2e9385ed4323e3576b99f89890affbf569e16afde
+  # sha256 is the preferred checksum -- you can get it for a file with:
+  #  `openssl sha256 <file name>`.
+  # You may need the openssl package, available on conda-forge:
+  #  `conda install openssl -c conda-forge``
+
+build:
```
Comment (xhochy) [inline recipes/sphinx-github-changelog/meta.yaml:26]:
```suggestion
build:
  noarch: python
```

**Review by xhochy — COMMENTED** [review]
Please remove all the example comments

**Comment by curtisma** [comment]
> * noarch: python recipes are required to have a lower bound on the python version. Typically this means putting `python >=3.6` in **both** `host` and `run` but you should check upstream for the package's Python compatibility.

@ewjoachim, do you know what's the minimum python version required?

**Comment by ewjoachim** [comment]
@xhochy Done :)

**Comment by ewjoachim** [comment]
Thank you !

**Pull request merged**